pkgs/development/python-modules/peblar/default.nix

```diff
@@ -0,0 +1,77 @@
+{
+  lib,
+  aiohttp,
+  aresponses,
+  awesomeversion,
+  backoff,
+  buildPythonPackage,
+  fetchFromGitHub,
+  hatchling,
+  mashumaro,
+  orjson,
+  pytest-asyncio,
+  pytest-cov-stub,
+  pytestCheckHook,
+  pythonOlder,
+  syrupy,
+  typer,
+  yarl,
+  zeroconf,
+}:
+
+buildPythonPackage rec {
+  pname = "peblar";
+  version = "0.3.3";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "frenck";
+    repo = "python-peblar";
+    tag = "v${version}";
+    hash = "sha256-Shusro3KspiMg44DaFW8sOoiRB0mvEZXh4x6/AmJXgQ=";
+  };
+
+  postPatch = ''
+    # Upstream doesn't set a version for GitHub releases
+    substituteInPlace pyproject.toml \
+      --replace-fail 'version = "0.0.0"' 'version = "${version}"'
+  '';
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    aiohttp
+    awesomeversion
+    backoff
+    mashumaro
+    orjson
+    yarl
+  ];
+
+  optional-dependencies = {
+    cli = [
+      typer
+      zeroconf
+    ];
```

Comment:
We probably want a top-level package (toPythonApplication) with those optional-dependencies  installed as well.

Reply:
We can add that if someone requests it. For now it seems unlikely this is used for anything but Home Assistant.